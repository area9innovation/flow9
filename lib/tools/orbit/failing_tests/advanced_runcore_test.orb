// Advanced RunCore test for more complex operations

// Test array operations with calculation
let arr1 = [10, 20, 30, 40, 50];
let idx = 2;
let val = index(arr1, idx);

// Test array subrange
let subarr = subrange(arr1, 1, 3);

// String index operations
let str = "Testing string indexing";
let idx_val = strIndex(str, 4);

// Numeric operations
let a = 15;
let b = 4;
let mod_res = a % b;  // Modulo operation

// Compare values
let cmp = a <=> b;  // Spaceship operator (-1, 0, 1)

// String/array operations
let words = ["This", "is", "a", "test"];
let joined = strGlue(words, " ");

// Test string contains at position
let contains_at = strContainsAt("HelloWorld", 5, "World");

// Test string to char array and back
let char_array = string2ints("ABC");
let str_back = ints2string(char_array);

// Test reversing an array
let reversed = reverse(arr1);

// Test string capitalization
let cap_str = capitalize("test string");
let decap_str = decapitalize("Test String");

// Test type checks
let is_int = isInt(a);
let is_str = isString(str);
let is_arr = isArray(arr1);

// Print results
let res = println("Array value at index " + i2s(idx) + ": " + i2s(val));
res = println("Subarray: " + i2s(subarr));
res = println("String character at index 4: " + idx_val);
res = println("Modulo result: " + i2s(mod_res));
res = println("Compare result: " + i2s(cmp));
res = println("Joined string: " + joined);
res = println("String contains at position: " + i2s(contains_at));
res = println("Character array: " + i2s(char_array));
res = println("String from char array: " + str_back);
res = println("Reversed array: " + i2s(reversed));
res = println("Capitalized string: " + cap_str);
res = println("Decapitalized string: " + decap_str);
res = println("Is integer check: " + i2s(is_int));
res = println("Is string check: " + i2s(is_str));
res = println("Is array check: " + i2s(is_arr));