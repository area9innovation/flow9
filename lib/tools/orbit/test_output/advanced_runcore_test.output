Flow compiler (3rd generation)

Processing 'orbit' on http server

tests/advanced_runcore_test.orb: tests/advanced_runcore_test.orb:21:14: Parse error
let cmp = if a > b then 1 else if a < b then -1 else 0;
             ^
