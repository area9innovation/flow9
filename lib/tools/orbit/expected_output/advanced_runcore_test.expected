Flow compiler (3rd generation)

Processing 'orbit' on http server

Array value at index 2: 30
Subarray: [20, 30, 40]
String character at index 4: i
Modulo result: 3
Compare result: 1
Joined string: This is a test
String contains at position: true
Character array: [65, 66, 67]
String from char array: ABC
Reversed array: [50, 40, 30, 20, 10]
Capitalized string: Test string
Decapitalized string: test String
Is integer check: true
Is string check: true
Is array check: true
Result: "Is array check: true"
VERIFY - THIS OUTPUT NEEDS HUMAN VERIFICATION
